packages/flutter_tools/lib/src/version.dart

```diff
@@ -285,33 +285,13 @@ class FlutterVersion {
       final String branch = _runGit('git rev-parse --abbrev-ref HEAD', globals.processUtils);
       return branch == 'HEAD' ? channel : branch;
     }();
-    if (redactUnknownBranches || _branch.isEmpty) {
+    if (redactUnknownBranches || _branch!.isEmpty) {
       // Only return the branch names we know about; arbitrary branch names might contain PII.
       if (!kOfficialChannels.contains(_branch)) {
         return '[user-branch]';
       }
     }
-    return _branch;
-  }
-
-  /// Returns true if `tentativeDescendantRevision` is a direct descendant to
-  /// the `tentativeAncestorRevision` revision on the Flutter framework repo
-  /// tree.
-  bool checkRevisionAncestry({
```

Comment:
This was dead.  It was introduced in https://github.com/flutter/flutter/pull/12959 to support a project migration that seems to have never happened.  It's [overridden but throws a `UnsupportedError` in g3](https://source.corp.google.com/piper///depot/google3/mobile/flutter/cli/lib/context.dart;l=448?q=tentativeDescendantRevision&sq=package:piper%20file:%2F%2Fdepot%2Fgoogle3%20-file:google3%2Fexperimental).

@xster as far as you know is this safe to delete?  We can pull it out of source control some day if https://github.com/flutter/flutter/issues/40460 gets done.

Reply:
Yes, it's a stub for a feature that was never implemented. Delete!